backend/data/src/main/kotlin/io/tolgee/component/fileStorage/S3FileStorageFactory.kt

```diff
@@ -8,6 +8,20 @@ class S3FileStorageFactory {
   fun create(config: S3Config): S3FileStorage {
     val client = S3ClientProvider(config).provide()
     val bucketName = config.bucketName ?: throw RuntimeException("Bucket name for S3 storage is not set")
-    return S3FileStorage(bucketName, client)
+    val path = getCanonicalPath(config.path)
+
+    return S3FileStorage(bucketName = bucketName, path = path, s3 = client)
+  }
+
+  private fun getCanonicalPath(path: String?): String {
```

Comment:
It's good you handled this, but...

- The code is very long and ugly (actually there already is a method for this)
   ```io.tolgee.fixtures.RemoveSlashSuffixKt#removeSlashSuffix```
- It does not handle the cases when the path is invalid
- It does not handle duplicated `/` at the end
- I am not sure whether it should be placed here. I would be afraid that someone will create the S3FileStorage will be created without this factory and so it would not handle the path correctly.